src/java.base/share/classes/sun/reflect/generics/reflectiveObjects/WildcardTypeImpl.java

```diff
@@ -77,6 +77,17 @@ public static WildcardTypeImpl make(FieldTypeSignature[] ubs,
         return new WildcardTypeImpl(ubs, lbs, f);
     }
 
+    /**
+     * Eager factory method.
+     * @param ubs - an array of types representing the upper bounds
+     * @param lbs - an array of types representing the lower bounds
+     * @return a wild card type with the requested bounds
+     */
+    public static WildcardTypeImpl make(Type[] ubs,
```

Comment:
I had to tweak this class: in our case we already know the type for the wildcard bounds, so we don't need to "resolve" them. The reflective code wants to push the bound parsing as lazily as possible, but this generality is not needed for us. Note that the existing code was already flexible enough to handle both the eager and the lazy case (e.g. lazy bounds are turned dynamically into proper bounds on first access), so the only thing left to do was to define an internal factory for the non-parsing flavor.